Enable CONF_STATION_UPDATES by default on AEMET

Noltari opened the pull request:
## Proposed change
The default state of CONF_STATION_UPDATES is enabled when it doesn't exist in the config:
https://github.com/home-assistant/core/blob/59066c1770d428ac51c20bf3fc29c8552a924d58/homeassistant/components/aemet/__init__.py#L32
However, the default config schema creates it as False:
https://github.com/home-assistant/core/blob/59066c1770d428ac51c20bf3fc29c8552a924d58/homeassistant/components/aemet/config_flow.py#L22-L26
This results in the checkbox not being enabled when a new config entry is created, despite being treated as enabled since it doesn't exist in the config:
![image](https://github.com/home-assistant/core/assets/264346/02648f18-1084-4c62-84c3-e7b581f67848)

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

Noltari committed:
aemet: set CONF_STATION_UPDATES default to True

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

thecode reviewed (approved):
LGTM, would be nice to add a test so someone doesn't break it in the future

Noltari committed:
tests: aemet: check CONF_STATION_UPDATES default form value

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari commented:
> LGTM, would be nice to add a test so someone doesn't break it in the future

@thecode done in f0f95da68fa5b2d8964b0a45cbd9939843deccfb, but I'm not sure if this is the proper way of doing it...

thecode commented:
> > LGTM, would be nice to add a test so someone doesn't break it in the future
> 
> @thecode done in [f0f95da](https://github.com/home-assistant/core/commit/f0f95da68fa5b2d8964b0a45cbd9939843deccfb), but I'm not sure if this is the proper way of doing it...

The best way to do it is to run the config flow and validate that the option is returned as true

Noltari commented:
> The best way to do it is to run the config flow and validate that the option is returned as true

Excuse me, but isn't that what I did?
Could you point me to some other test which does something similar so that I can use it as a guide?

thecode commented:
> > The best way to do it is to run the config flow and validate that the option is returned as true
> 
> Excuse me, but isn't that what I did? Could you point me to some other test which does something similar so that I can use it as a guide?

First I am not an expert in this :-), but I was thinking something like this:
Call the bellow with an empty `user_input`
https://github.com/home-assistant/core/blob/f0f95da68fa5b2d8964b0a45cbd9939843deccfb/tests/components/aemet/test_config_flow.py#L101-L103

validate that the retuned value is true:
https://github.com/home-assistant/core/blob/f0f95da68fa5b2d8964b0a45cbd9939843deccfb/tests/components/aemet/test_config_flow.py#L119-L121

Maybe you can even parametrize the test to run twice, once with `False` as the user input and once without empty user input.

Noltari committed:
Revert "tests: aemet: check CONF_STATION_UPDATES default form value"

This reverts commit f0f95da68fa5b2d8964b0a45cbd9939843deccfb.

Noltari committed:
tests: aemet: parametriz test_form_options

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari commented:
> First I am not an expert in this :-), but I was thinking something like this: Call the bellow with an empty `user_input`
> 
> https://github.com/home-assistant/core/blob/f0f95da68fa5b2d8964b0a45cbd9939843deccfb/tests/components/aemet/test_config_flow.py#L101-L103
> 
> validate that the retuned value is true:
> 
> https://github.com/home-assistant/core/blob/f0f95da68fa5b2d8964b0a45cbd9939843deccfb/tests/components/aemet/test_config_flow.py#L119-L121
> 
> Maybe you can even parametrize the test to run twice, once with `False` as the user input and once without empty user input.

Thanks for the help @thecode,
I've reverted f0f95da68fa5b2d8964b0a45cbd9939843deccfb in 0060b83fdacd83d1ea09d5df120b56f08cf68dda.
And implemented your suggestions on e628f4164c29ec6b9f068510e1af41defdf47239

thecode reviewed (approved):
Thanks!
